src/test/java/org/opensearch/knn/index/VectorDataTypeIT.java

```diff
@@ -176,6 +190,202 @@ public void testByteVectorDataTypeWithNmslibEngine() {
+
+        validateL2SearchResults(response);
+    }
+
+    public void testL2PainlessScriptingWithFloatVectorDataType() throws Exception {
+        createKnnIndexMappingForScripting(2, VectorDataType.FLOAT.getValue());
+        ingestL2FloatTestData();
+
+        String source = String.format("1/(1 + l2Squared([1.0f, 1.0f], doc['%s']))", FIELD_NAME);
+        Request request = constructScriptScoreContextSearchRequest(
+            INDEX_NAME,
+            MATCH_ALL_QUERY_BUILDER,
+            Collections.emptyMap(),
+            Script.DEFAULT_SCRIPT_LANG,
+            source,
+            4
+        );
+
+        Response response = client().performRequest(request);
+        assertEquals(request.getEndpoint() + ": failed", RestStatus.OK, RestStatus.fromCode(response.getStatusLine().getStatusCode()));
+
+        validateL2SearchResults(response);
+    }
+
+    public void testKNNScriptScoreWithInvalidVectorDataType() {
+        // Set an invalid value for data_type field while creating the index for script scoring which should throw an exception
+        ResponseException ex = expectThrows(ResponseException.class, () -> createKnnIndexMappingForScripting(2, "invalid_data_type"));
+        assertTrue(
+            ex.getMessage()
+                .contains(
+                    String.format(
+                        Locale.ROOT,
+                        "Invalid value provided for [%s] field. Supported values are [%s]",
+                        VECTOR_DATA_TYPE_FIELD,
+                        SUPPORTED_VECTOR_DATA_TYPES
+                    )
+                )
+        );
+    }
+
+    public void testKNNScriptScoreWithInvalidByteQueryVector() throws Exception {
+        // Create an index with byte vector data_type, add docs and run a scoring script query with decimal values
+        // which should throw exception
+        createKnnIndexMappingForScripting(2, VectorDataType.BYTE.getValue());
+
+        Byte[] f1 = { 6, 6 };
+        addKnnDoc(INDEX_NAME, "1", FIELD_NAME, f1);
+
+        Byte[] f2 = { 2, 2 };
+        addKnnDoc(INDEX_NAME, "2", FIELD_NAME, f2);
+
+        // Construct Search Request with query vector having decimal values
+        Float[] queryVector = { 10.67f, 19.78f };
+        Request request = createScriptQueryRequest(queryVector, SpaceType.L2.getValue(), MATCH_ALL_QUERY_BUILDER);
+        ResponseException ex = expectThrows(ResponseException.class, () -> client().performRequest(request));
+        assertTrue(
+            ex.getMessage()
+                .contains(
+                    String.format(
+                        Locale.ROOT,
+                        "[%s] field was set as [%s] in index mapping. But, KNN vector values are floats instead of byte integers",
+                        VECTOR_DATA_TYPE_FIELD,
+                        VectorDataType.BYTE.getValue()
+                    )
+                )
+        );
+    }
+
+    @SneakyThrows
+    private void ingestL2ByteTestData() {
```

Comment:
Why does this need to be L2?

Reply:
I actually thought of creating different methods for test data for other spacetypes later. So, I created it with L2 as I'm using it for spacetype L2.